enumerable.rb

```diff
@@ -0,0 +1,102 @@
+# Custom Enumerable Methods
+module ::Enumerable
+  # Calls block with the item for each item in enum.
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |n| yield self[n] }
+    self
+  end
+
+  # Loops, Calling The Block With Item And Index
+  def my_each_with_index(&block)
+    for index in 0...count
+      block.call(self[index], index)
+    end
+  end
+
+  # Returns an array containing all elements of enum for which the given block returns a true value.
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each do |n|
+      array << n if yield n
+    end
+    array
+  end
+
+  # Passes each element of the collection to the given block.
+  # The method returns true if the block never returns false or nil.
+  def my_all?(&block)
+    my_each do |item|
+      return false if block.call(item) != true
+    end
+
+    true
+  end
```

Comment:
### #my_all?
Right now my_all is not behaving as it should, please refer to this [doc](https://apidock.com/ruby/Enumerable/all%3F)

please make sure it behaves as same as #all? method before submitting for a new review. right now is falling when:
- [ ] self is a range,
- [ ] The collection has a boolean or nil element
- [ ] The collection element is compared with a class
- [ ] The collection element is tested with a regex

### [__Optional__]
Please test all of your code by comparing it with the #all? method before submitting for a new review.

Reply:
Fixed In 
https://github.com/epinczinger/ruby_enumerables/pull/7/commits/e93ae33bfe630a305f6e2759cc4811db5b4db608
https://github.com/epinczinger/ruby_enumerables/pull/7/commits/04c35a8397b0b727cd30db919d0456fb0dbb8381
https://github.com/epinczinger/ruby_enumerables/pull/7/commits/cf1ac7bc16af12a3d6445720a2cb222f875f8a9b